src/main/java/ev3dev/sensors/ev3/EV3GyroSensor.java

```diff
@@ -118,4 +121,53 @@ public void reset() {
         switchMode(MODE_RATE_ANGLE, SWITCH_DELAY);
     }
 
+    public static class EV3GyroAngleMode implements SensorMode {
```

Comment:
That mode is not necessary. I understand that you added the method getAngle() but it breaks the contract.
It is necessary to upgrade with DataChannelRereader: 
https://github.com/ev3dev-lang-java/ev3dev-lang-java/blob/master/src/main/java/ev3dev/sensors/GenericMode.java

Reply:
In what way does it break the contract?

Touching GenericMode is a much broader change. I'll put in a pull request to show that.